worlds/sm/variaRandomizer/utils/objectives.py

```diff
@@ -511,16 +511,18 @@ def updateItemPercentEscapeAccess(self, collectedLocsAccessPoints):
     def setScavengerHuntFunc(self, scavClearFunc):
         self.goals["finish scavenger hunt"].clearFunc = scavClearFunc
 
-    def setItemPercentFuncs(self, totalItemsCount=None, allUpgradeTypes=None):
-        def getPctFunc(pct, totalItemsCount):
+    def setItemPercentFuncs(self, totalItemsCount=None, allUpgradeTypes=None, container=None):
+        def getPctFunc(pct, totalItemsCount, container):
             def f(sm, ap):
-                nonlocal pct, totalItemsCount
-                return sm.hasItemsPercent(pct, totalItemsCount)
+                nonlocal pct, totalItemsCount, container
+                locs_checked = len(container.getUsedLocs(lambda loc: True))
+                return SMBool(locs_checked * 100 / totalItemsCount >= pct)
```

Comment:
```suggestion
                locs_checked = len(container.getUsedLocs(lambda loc: True))
                return SMBool(locs_checked >= total_needed)
```

Reply:
calculating it once ahead of the function should make it a nice bit faster.

Reply:
Fixed with latest commit.